akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+to :class:`PersistentActor`` s and persistence queries, which allows these layers to focus on the business problems instead
+handling the different schemas explicitly in the business logic layers.
+
+
+In summary, schema evolution in event sourced systems exposes the following characteristics:
+  - Allow the system to continue operating without large scale migrations to be applied,
+  - Allow the system to read "old" events from the underlying storage, however present them in a "new" view to the application logic,
+  - Transparently promote events to the latest versions during recovery (or queries) such that the business logic need not consider multiple versions of events
+
+.. _Immutability Changes Everything: http://www.cidrdb.org/cidr2015/Papers/CIDR15_Paper16.pdf
+.. _Event Sourcing: http://martinfowler.com/eaaDev/EventSourcing.html
+
+Picking the right serialization format
+======================================
+
+Picking the serialization format is one of the most important decisions you will be making in setting up
+your project. It affects which kind of evolutions are simple (or hard) to do, how much work is required to
+add a new datatype and last but not least performance serialization.
+
+Provided default serializers
+----------------------------
+
+Akka Persistence provides `Google Protocol Buffers`_ based serializers (using :ref:`Akka Serialization <serialization-scala>`)
+for it's own message types such as ``PersistentRepr``, ``AtomicWrite`` and snapshots. Journal plugin implementations
+*may* choose to use those provided serializers, or pick a serializer which suits the underlying database more.
+
+.. note::
+  Serialization is **NOT** handled automatically by Akka Persistence itself. Instead, it only provides the above described
+  serializers, and in case a ``AsyncWriteJournal`` plugin implementation chooses to use them directly, the above serialization
+  scheme will be used.
+
+  Please refer to your write journal's documentation to learn more how it handles serialization!
+
+  For example, some journals may choose to not use Akka Serialization *at all* and instead store the data in a format
+  that is more "native" for the underlying datastore, e.g. using JSON or some other kind of format that the target
+  datastore understands natively.
+
+The below figure explains how the default serialization scheme works, and how it fits together with serializing the
+user provided message itself, which we will from here on refer to as the ``payload`` (highlighted in yellow):
+
+.. figure:: ../images/persistent-message-envelope.png
+   :align: center
+
+   Akka Persistence provided serializers wrap the user payload in an envelope containing all persistence-relevant information.
+   **The payload will be serialized using the user configured serializer (if none is provided explicitly, Java serialization will be used).**
```

Comment:
Thanks, I'll reword the entire section to explain more explicitly where which serializer is used.